ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -107,6 +119,7 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
    * @return the list of {@link NetworkSend} objects to hand over to the Selector.
    */
   private List<NetworkSend> prepareSends(List<ResponseInfo> responseInfoList) {
+    long prepareSendStartTimeInMs = System.currentTimeMillis();
```

Comment:
again, I think we have to be careful about these metrics to prevent metric overload. We don't really want to instrument every single method for prod code - there should be a reason. These methods do not do any I/O or cpu intensive stuff. For all you know they finish under a millisecond. If we make a case for it here, we could make a case for doing this for every single non-trivial method in ambry.

We could add a metric at a higher level - how long `sendAndPoll` takes, for example, how long operation manager `poll`s take etc. but I don't see a need for having metrics at this level.